browser-switch/src/test/java/com/braintreepayments/api/BrowserSwitchListenerFinderUnitTest.java

```diff
@@ -0,0 +1,94 @@
+package com.braintreepayments.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class BrowserSwitchListenerFinderUnitTest {
+
+    private static class ListenerActivity extends FragmentActivity implements BrowserSwitchListener {
+        @Override
+        public void onBrowserSwitchResult(BrowserSwitchResult result) {
+        }
+    }
+
+    private static class ListenerFragment extends Fragment implements BrowserSwitchListener {
+        @Override
+        public void onBrowserSwitchResult(BrowserSwitchResult result) {
+        }
+    }
+
+    private FragmentActivity plainActivity;
```

Comment:
```suggestion
    private FragmentActivity activity;
```

Reply:
I agree the variable naming isn't clear here. I went with `nonListenerActivity` instead to show that it should skip notifying a `FragmentActivity` that doesn't implment `BrowserSwitchListener`.